aten/src/ATen/native/quantized/cpu/qconv_dynamic.cpp

```diff
@@ -0,0 +1,184 @@
+#include <algorithm>
+#include <cmath>
+#include <vector>
+
+#include <ATen/ATen.h>
+#include <ATen/Parallel.h>
+#include <ATen/SmallVector.h>
+#include <ATen/native/quantized/cpu/conv_packed_params.h>
+#include <ATen/native/quantized/cpu/fbgemm_utils.h>
+#include <ATen/native/quantized/cpu/qnnpack_utils.h>
+#include <ATen/native/quantized/cpu/quant_utils.h>
+#include <c10/util/irange.h>
+#include <caffe2/utils/threadpool/pthreadpool-cpp.h>
+#include <torch/library.h>
+
+#ifdef USE_FBGEMM
+
+template <int kSpatialDim>
+at::Tensor PackedConvWeight<kSpatialDim>::apply_dynamic(
+    const at::Tensor& input,
+    bool reduce_range) {
+  /// is this needed?
+  auto input_contig = input.contiguous();
```

Comment:
This should be channels last contiguous. fbgemm expects input to be in channels last format. Check the equivalent qconv implementation.

Reply:
Actually if we want this to work for transposed conv as well, I think it is better to just not call `contiguous` here. For doing per tensor quantization I dont think it matters.